File: server/src/main/java/org/cleanas2/bus/MessageBase.java
package org.cleanas2.bus;

/**
 * @author Andrew Backer {@literal awbacker@gmail.com / andrew.backer@powere2e.com}
 */
public class MessageBase {
    private boolean error = false;
    private Exception errorCause = null;

    public boolean isError() {
        return error;
    }

    public void setError(boolean error) {
        this.error = error;
    }

    public Exception getErrorCause() {
        return errorCause;
    }

    public void setErrorCause(Exception errorCause) {
        this.setError(true);
        this.errorCause = errorCause;
    }
}

File: server/src/main/java/org/cleanas2/bus/MessageBase.kt
package org.cleanas2.bus

/**
 * @author Andrew Backer awbacker@gmail.com / andrew.backer@powere2e.com
 */
open class MessageBase {
    var isError = false
    var errorCause: Exception? = null
        set(errorCause) {
            this.isError = true
            field = errorCause
        }
}
